yapapi/engine.py

```diff
@@ -459,40 +459,49 @@ async def _process_invoices(self) -> None:
 
     async def _process_debit_notes(self) -> None:
         """Process incoming debit notes."""
+        debit_note_processing_tasks = []
+        loop = asyncio.get_event_loop()
 
-        async for debit_note in self._payment_api.incoming_debit_notes():
-            job_id = next(
-                (
-                    id
-                    for id in self._agreements_accepting_debit_notes
-                    if debit_note.agreement_id in self._agreements_accepting_debit_notes[id]
-                ),
-                None,
+        async for debit_note_id in self._payment_api.incoming_debit_note_ids():
+            debit_note_processing_tasks.append(
```

Comment:
I would add a semaphore and solve it like here:
https://github.com/golemfactory/yapapi/blob/81730767217381cb905e1a5ba869ba80ea4f82f0/yapapi/engine.py#L806